### PR TITLE
loading spinner, Progress Dialog and movie details button animate

### PR DESCRIPTION
**Pull request by candry7731**
Added loading spinner to movies detail screens

Added loading spinner to the movie details screen

Added progress dialog when user clicks the play button for movies and trailers

Updated movie details buttons to be hidden until all data is loaded

Updated critic rating star to be hidden until all data is loaded

### CODE REVIEW

**Comment by jimdogx** [comment]
I noticed that when I hit Play Movie, I get the new "Loading Movie" spinner, that goes away and then I get the regular spinner (not text).  Not sure if that was the desired behavior?

**Comment by whiteowl3** [comment]
If the server holding the strm fails to respond, the spinner locks us out forever.  back or instant replay should cancel the loading job and clear the spinner.
Why it is so visually different from all the other spinners?

**Comment by whiteowl3** [comment]
> I noticed that when I hit Play Movie, I get the new "Loading Movie" spinner, that goes away and then I get the regular spinner (not text). Not sure if that was the desired behavior?

The first spinner is to lock out additional inputs (esp play inputs) until jellyfin responds with the stream.  if the media is remote, this can take a second.  so, the first spinner drops as soon as jellyfin says "here it is" and the second spinner waits until we have enough buffered to play.  we need both.

**Comment by candry7731** [comment]
This should be good to go unless anyone else finds any bugs.

**Comment by 1hitsong** [comment]
This works as described for me. 

@jimdogx You good?

@candry7731 Can you resolve the merge conflict?

**Comment by jimdogx** [comment]
This might be due to the conflicts, but if I run this branch on a movie that is partially watched (so I should see the "Resume" or "start over" popup), I am shown "Loading Movie" with a spinner that never goes away.

**Comment by 1hitsong** [comment]
Status: Pending code updates.

**Comment by candry7731** [comment]
> This might be due to the conflicts, but if I run this branch on a movie that is partially watched (so I should see the "Resume" or "start over" popup), I am shown "Loading Movie" with a spinner that never goes away.

This should be fixed. Please test and let me know.

**Comment by sevenrats** [comment]
it's important that we restart the progress dialog in VideoPlayer.brs after we have closed it in startPlayBackOver.

**Comment by 1hitsong** [comment]
From Home View, Click TV Episode from Continue Watching
- [x] From popup click Go To Series. Spinner shows and doesn't go away
- [x] From popup click Go To Season. Spinner shows and doesn't go away
- [x] From popup click Go To Episode. Spinner shows and doesn't go away

**Comment by candry7731** [comment]
I'm also getting an error when clicking on the season...  

image

**Comment by candry7731** [comment]
I'm working on these now.... Just fyi

**Comment by candry7731** [comment]
These are completed and ready for review

File: components/movies/MovieDetails.xml [hunk 1]
```diff
@@ -29,7 +29,7 @@
           <Label id="audio_codec" vertAlign="bottom" height="39" />
           <label id="audio_codec_count" font="font:smallestSystemFont" vertAlign="top" color="#ceffff" />
         </LayoutGroup>
-        <ButtonGroupHoriz id="buttons" itemSpacings="[10]">
+        <ButtonGroupHoriz id="buttons" itemSpacings="[10]" visible = "false">
```
Comment (1hitsong) [inline components/movies/MovieDetails.xml]:
```suggestion
        <ButtonGroupHoriz id="buttons" itemSpacings="[10]" visible="false">
```

File: components/movies/MovieDetails.xml [hunk 2]
```diff
@@ -41,6 +41,8 @@
       </LayoutGroup>
     </LayoutGroup>
     <MovieOptions id="movieOptions" visible="false" />
+    <Spinner id="spinner" translation="[900, 450]" visible = "true" />
```
Comment (1hitsong) [inline components/movies/MovieDetails.xml]:
```suggestion
    <Spinner id="spinner" translation="[900, 450]" visible="true" />
```

File: source/Main.brs [hunk 3]
```diff
@@ -411,6 +411,9 @@ sub Main (args as dynamic) as void
                 end if
 
             else if btn <> invalid and btn.id = "trailer-button"
+                dialog = createObject("roSGNode", "ProgressDialog")
+                dialog.title = tr("Loading trailer")
```
Comment (1hitsong) [inline source/Main.brs:415]: "Loading trailer" needs to be added to translations

**Review by 1hitsong — CHANGES_REQUESTED** [review]
Functionality looks good. Just a few small finishing touches and it's done.

File: source/Main.brs (hunk 3 above)
Comment (candry7731) [reply source/Main.brs:415]: Done

**Review by 1hitsong — APPROVED** [review]
To be merged 2/12 unless an issue arises.

**Pull request merged**